### DISCUSSION TITLE
API question: setValue isn't triggering form change

### DISCUSSION THREAD

**Discussion by r04423** [react-hook-form/react-hook-form#3050] [Q&A]
Hi! 👋

Let's say I have a `<form>` with an `onChange` callback where I do some business logic with the data.
In this form I also have a controlled input (3rd-party library) which is working absolutely fine in the context of the library, however it does not trigger form's `onChange` event which I mentioned before.

Just wanted to ask - is it intended?
If not, do you think it would be a nice feature to add? Or did I miss something in the documentation that could help me solve this problem?

Thanks a ton!

**Comment by bluebill1049** [comment]
`setValue` doesn't trigger `onChange`, what's the use case you would like to trigger `onChange`?

**Reply by r04423** [reply]
I'm using `react-hook-form` for a side filter which doesn't have a typical submit button, thus form is submitted on every value change (by using `handleSubmit` on `onChange` instead of `onSubmit`).

Form's values are typically radio / checkbox groups, however there's also a `DateRangePicker` (this 3rd-party library) which is wrapped in `<Controller>`.
After selecting the value in the calendar the input's value changes correctly and so the form's state, however the form's `onChange`  event is not triggered.

As a workaround I currently `watch` field's value and manually trigger `onChange` in an `useEffect` hook, however it seemed to me that it is a feature that could be in the library itself.
